src/mscorlib/src/System/AppContext/AppContext.cs

```diff
@@ -65,79 +64,83 @@ public static bool TryGetSwitch(string switchName, out bool isEnabled)
-                    //    Example: TryGetSwitch is called for a switch set via SetSwitch
-                    // 3. The switch has the default value and we need to check for overrides
-                    //    Example: TryGetSwitch is called for the first time for a switch that has a default value 
-
-                    // 1. The value is unknown
-                    if (switchValue == SwitchValueState.UnknownValue)
-                    {
-                        isEnabled = false;
-                        return false;
-                    }
-
-                    // We get the value of isEnabled from the value that we stored in the dictionary
-                    isEnabled = (switchValue & SwitchValueState.HasTrueValue) == SwitchValueState.HasTrueValue; 
-
-                    // 2. The switch has a valid value AND we have checked for overrides
-                    if ((switchValue & SwitchValueState.HasLookedForOverride) == SwitchValueState.HasLookedForOverride)
-                    {
-                        return true;
-                    }
-
-                    // 3. The switch has a valid value, but we need to check for overrides.
-                    // Regardless of whether or not the switch has an override, we need to update the value to reflect
-                    // the fact that we checked for overrides. 
-                    bool overrideValue;
-                    if (AppContextDefaultValues.TryGetSwitchOverride(switchName, out overrideValue))
-                    {
-                        // we found an override!
-                        isEnabled = overrideValue;
-                    }
+                    isEnabled = false;
+                    return false;
+                }
 
-                    // Update the switch in the dictionary to mark it as 'checked for override'
-                    s_switchMap[switchName] = (isEnabled ? SwitchValueState.HasTrueValue : SwitchValueState.HasFalseValue)
-                                                | SwitchValueState.HasLookedForOverride;
+                // We get the value of isEnabled from the value that we stored in the dictionary
+                isEnabled = (switchValue & SwitchValueState.HasTrueValue) == SwitchValueState.HasTrueValue; 
 
+                // 2. The switch has a valid value AND we have checked for overrides
+                if ((switchValue & SwitchValueState.HasLookedForOverride) == SwitchValueState.HasLookedForOverride)
+                {
                     return true;
                 }
-                else
+
+                // 3. The switch has a valid value, but we need to check for overrides.
+                // Regardless of whether or not the switch has an override, we need to update the value to reflect
+                // the fact that we checked for overrides. 
+                bool overrideValue;
+                if (AppContextDefaultValues.TryGetSwitchOverride(switchName, out overrideValue))
```

Comment:
Without the lock, this code could end up executing more than once. I think this should stay inside the lock as the call it makes is expensive and we should try and only do it once.